src/app/cesium.service.ts

```diff
@@ -509,6 +514,58 @@ export class CesiumService {
   }
 
 
+  public getAndLoadTfrNoFlyZones(): void {
+    console.log('Loading TFR No FlyZones');
+
+    this.httpClient
+      .get<GetNoFlyZonesResponse>(
+        Url.consumer('/getTfrNoFlyZones'),
+        this.httpOptions
+      )
+      .subscribe((data) => {
```

Comment:
Angular rxjs has a built-in lastValueFrom and firstValueFrom that allows to take advantage of async await and reduce nesting.